### PR TITLE
Added mathematical description to Noisy SGD

### PR DESCRIPTION
**Pull request by hmludwig**
I added mathematical description to noisy SGD to improve clarity (see #757)

The results look like this:
Screenshot 2024-03-08 at 12 46 34

### CODE REVIEW

**Review by fabianp — APPROVED** [review]
looks great to me, thanks!

File: optax/_src/alias.py [hunk 1]
```diff
@@ -992,11 +992,27 @@ def noisy_sgd(
     gamma: float = 0.55,
     seed: int = 0
 ) -> base.GradientTransformation:
-  r"""A variant of SGD with added noise.
```
Comment (fabianp) [inline optax/_src/alias.py:995]: First line needs to end with ".", so I would leave the first line as it is, and add your description below

Comment (fabianp) [reply optax/_src/alias.py:995]: and the line after should be blank, so something like:

```
r"""A variant of SGD with added noise.

Noisy SGD is ...
"""

Comment (hmludwig) [reply optax/_src/alias.py:995]: Could you clarify if the requirement for the first line to end with "." is driven by stylistic guidelines or technical reasons? The list of optimizers at the beginning of the documention isn't effected and displays the first sentence of the docstring correctly.

Comment (fabianp) [reply optax/_src/alias.py:995]: its a requirement of the python style: https://peps.python.org/pep-0257/#multi-line-docstrings

Internally we get an error when importing the code if the docstring doesn't follow this convention

Comment (fabianp) [reply optax/_src/alias.py:995]: you're right though that this would be something worth adding to https://optax.readthedocs.io/en/latest/development.html

File: optax/_src/alias.py [hunk 2]
```diff
@@ -994,9 +994,27 @@ def noisy_sgd(
...
-  
+  Noisy SGD is a variant of :func:`optax.sgd` that incorporates Gaussian noise
+  into the updates. It has been found that adding noise to the gradients can
+  improve both the training error and the generalization error in very deep
+  networks.
+
+  The update :math:`u_t` is modified to include this noise as follows:
+
+  .. math::
+    u_t \leftarrow -\alpha_t g_t + N(0, \sigma_t^2),
```
Comment (vroulet) [inline optax/_src/alias.py:1005]: Thanks for doing that!
Shouldn't it rather be the following?
`u_t \leftarrow -\alpha_t (g_t + N(0, \sigma_t^2))`

**Review by vroulet — COMMENTED** [review]
Thanks for doing this! 
As you are on it, could you change the description of `gamma` lines 1046-1047 of this file by 
```
gamma: A parameter controlling the annealing of noise over time ``t``, the
      variance decays according to ``(1+t)**(-gamma)``.
```
Thank you again!

File: optax/_src/alias.py (hunk 2 above)
Comment (hmludwig) [reply optax/_src/alias.py:1005]: Yes, you're right. We first add the noise to the gradient and then scale by the learning rate. I will review my more carefully in the future.
